src/Stratis.Bitcoin.Features.MemoryPool/MempoolOrphans.cs

```diff
@@ -151,17 +152,21 @@ public class OrphanTx
                     // might be now valid, e.g. due to a nLockTime'd tx becoming valid,
                     // or a double-spend. Reset the rejects filter and give those
                     // txs a second chance.
+                    this.logger.LogTrace("Executing task to clear rejected transactions.");
                     this.hashRecentRejectsChainTip = this.chain.Tip.HashBlock;
                     this.recentRejects.Clear();
                 });
-            }
+            }            
 
             // Use pcoinsTip->HaveCoinsInCache as a quick approximation to exclude
             // requesting or processing some txs which have already been included in a block
-            return await this.MempoolLock.ReadAsync(() =>
+            bool isTxPresent = await this.MempoolLock.ReadAsync(() =>
```

Comment:
I think you only need the mempool lock for this `this.memPool.Exists(trxid)` and this should be available on the manager.